src/overlays/actors/ovl_En_Shopnuts/z_en_shopnuts.c

```diff
@@ -30,41 +37,265 @@ const ActorInit En_Shopnuts_InitVars = {
+    Collider_CylinderUpdate(&this->actor, &this->collider);
+
+    if (((this->actor.params == 0x0002) && (gSaveContext.itemGetInf[0] & 0x800)) ||
+        ((this->actor.params == 0x0009) && (gSaveContext.infTable[25] & 4)) ||
+        ((this->actor.params == 0x000A) && (gSaveContext.infTable[25] & 8))) {
+        Actor_Kill(&this->actor);
+    } else {
+        EnShopnuts_SetupWait(this);
+    }
+}
+
+void EnShopnuts_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnShopnuts* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void EnShopnuts_SetupWait(EnShopnuts* this) {
+    Animation_PlayOnceSetSpeed(&this->skelAnime, &D_0600139C, 0.0f);
+    this->animFlagAndTimer = Rand_S16Offset(100, 50);
+    this->collider.dim.height = 5;
+    this->collider.base.acFlags &= ~0x1;
+    this->actionFunc = EnShopnuts_Wait;
+}
+
+void EnShopnuts_SetupLookAround(EnShopnuts* this) {
+    Animation_PlayLoop(&this->skelAnime, &D_06000BA0);
+    this->animFlagAndTimer = 2;
+    this->actionFunc = EnShopnuts_LookAround;
+}
+
+void EnShopnuts_SetupThrowNut(EnShopnuts* this) {
+    Animation_PlayOnce(&this->skelAnime, &D_060001EC);
+    this->actionFunc = EnShopnuts_ThrowNut;
+}
+
+void EnShopnuts_SetupStand(EnShopnuts* this) {
+    Animation_MorphToLoop(&this->skelAnime, &D_06004574, -3.0f);
+    if (this->actionFunc == EnShopnuts_ThrowNut) {
+        this->animFlagAndTimer = 2 | 0x1000; // sets timer and flag
+    } else {
+        this->animFlagAndTimer = 1;
+    }
+    this->actionFunc = EnShopnuts_Stand;
+}
+
+void EnShopnuts_SetupBurrow(EnShopnuts* this) {
+    Animation_MorphToPlayOnce(&this->skelAnime, &D_0600039C, -5.0f);
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_NUTS_DOWN);
+    this->actionFunc = EnShopnuts_Burrow;
+}
+
+void EnShopnuts_SetupSpawnSalesman(EnShopnuts* this) {
+    Animation_MorphToPlayOnce(&this->skelAnime, &D_06000764, -3.0f);
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_NUTS_DAMAGE);
+    this->collider.base.acFlags &= ~0x1;
+    this->actionFunc = EnShopnuts_SpawnSalesman;
+}
+
+void EnShopnuts_Wait(EnShopnuts* this, GlobalContext* globalCtx) {
+    s32 hasSlowPlaybackSpeed;
+
+    hasSlowPlaybackSpeed = 0;
+    if (this->skelAnime.playSpeed < 0.5f) {
+        hasSlowPlaybackSpeed = 1;
+    }
+    if (hasSlowPlaybackSpeed) {
+        DECR(this->animFlagAndTimer);
+    }
+    if (Animation_OnFrame(&this->skelAnime, 9.0f) != 0) {
+        this->collider.base.acFlags |= 1;
+    } else if (Animation_OnFrame(&this->skelAnime, 8.0f) != 0) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_NUTS_UP);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Shopnuts/func_80AFA880.s")
+    this->collider.dim.height = ((CLAMP(this->skelAnime.curFrame, 9.0f, 13.0f) - 9.0f) * 9.0f) + 5.0f;
+    if (!hasSlowPlaybackSpeed && (this->actor.xzDistFromLink < 120.0f)) {
+        EnShopnuts_SetupBurrow(this);
+    } else if (SkelAnime_Update(&this->skelAnime) != 0) {
```

Comment:
```suggestion
    } else if (SkelAnime_Update(&this->skelAnime)) {
```
as does `SkelAnime_Update`